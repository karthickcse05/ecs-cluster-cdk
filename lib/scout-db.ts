import * as cdk from '@aws-cdk/core';
import * as ecs from '@aws-cdk/aws-ecs';
import * as ec2 from '@aws-cdk/aws-ec2';
import * as elbv2 from '@aws-cdk/aws-elasticloadbalancingv2';
import * as iam from '@aws-cdk/aws-iam';
import * as ssl_cert from '@aws-cdk/aws-certificatemanager';
import * as parameters from "../cdk.json";
import { LogGroup, LogStream } from "@aws-cdk/aws-logs";
import * as ssm from '@aws-cdk/aws-ssm';
import * as ecr from "@aws-cdk/aws-ecr"
import { BaseSecurityGroup } from './constructs/BaseSecurityGroup';
import { Constants } from './constants/constants';

export class ScoutDBStack extends cdk.Stack {

  constructor(scope: cdk.Construct, id: string, props?: cdk.StackProps) {
    super(scope, id, props);

    const defaultVpc = ec2.Vpc.fromLookup(this, 'ScoutDefaultVpc-DB', {
      isDefault: true,
    });

    // const postgrestdbSG = new ec2.SecurityGroup(this, "ScoutDB-SG-BackEnd", {
    //   vpc: defaultVpc,
    //   allowAllOutbound: true,
    //   securityGroupName: parameters.context.config.ProjectName + '-' + parameters.context.config.ProjectEnvironment + '-' + 'db-sg',
    // });

    const postgrestdbSG = new BaseSecurityGroup(this, "ScoutDB-SG", {
      vpc:defaultVpc,
      sgPropsName:"ScoutDB-SG-BackEnd",
      sgName: parameters.context.config.ProjectName + '-' + parameters.context.config.ProjectEnvironment + '-' + 'db-sg',
      allowAllOutbound: true,
      sgPortInfo: [
        {
          sgPort : Constants.securityGrouptcpPort,
          ipv4Port: '128.229.4.0/24',
          tcpPort:5432,
          description:Constants.securityGroupAllowHttpsDesc
        },
        {
          sgPort : Constants.securityGrouptcpPort,
          ipv4Port: '108.48.153.254/32',
          tcpPort:5432,
          description:Constants.securityGroupAllowHttpsDesc
        },
        {
          sgPort : Constants.securityGrouptcpPort,
          ipv4Port: '136.226.18.0/24',
          tcpPort:5432,
          description:Constants.securityGroupAllowHttpsDesc
        },
        {
          sgPort : Constants.securityGrouptcpPort,
          ipv4Port: '156.80.4.0/24',
          tcpPort:5432,
          description:Constants.securityGroupAllowHttpsDesc
        }
      ]
    });

    // postgrestdbSG.addIngressRule(
    //   ec2.Peer.ipv4('128.229.4.0/24'),
    //   ec2.Port.tcp(5432),
    //   "Allow https traffic"
    // );
    // postgrestdbSG.addIngressRule(
    //   ec2.Peer.ipv4('108.48.153.254/32'),
    //   ec2.Port.tcp(5432),
    //   "Allow https traffic"
    // );
    // postgrestdbSG.addIngressRule(
    //   ec2.Peer.ipv4('128.229.67.14/32'),
    //   ec2.Port.tcp(5432),
    //   "Allow https traffic"
    // );
    // postgrestdbSG.addIngressRule(
    //   ec2.Peer.ipv4('136.226.18.0/24'),
    //   ec2.Port.tcp(5432),
    //   "Allow https traffic"
    // );
    // postgrestdbSG.addIngressRule(
    //   ec2.Peer.ipv4('156.80.4.0/24'),
    //   ec2.Port.tcp(5432),
    //   "Allow https traffic"
    // );
    const importedSG = cdk.Fn.importValue("ecssg");
    //Create an ECS Cluster for back end
    const scoutdb_cluster = ecs.Cluster.fromClusterAttributes(this, 'ScoutDB-Imported-Cluster', {
      clusterName: parameters.context.config.ScoutappContainername + '-' + parameters.context.config.ProjectEnvironment + '-' + parameters.context.config.clustername,
      securityGroups: [ec2.SecurityGroup.fromSecurityGroupId(this, 'Imported DB SG', importedSG)],
      vpc: defaultVpc
    });

    //Task Execution role for Frond end Task Definition
    const ecsTaskExecutionRole = iam.Role.fromRoleArn(this, 'ImportedEcsTaskExecutionRole', parameters.context.config.ecsTaskExecutionRoleArn, {
      mutable: false
    })
    //const ecsTaskExecutionRole = iam.Role.fromRoleArn(this, 'Imported-Ecs-Task--Execution-Role', `arn:aws:iam::${parameters.context.config.Account_ID}:role/ecsTaskExecutionRole`)

    //Creation of Execution role for Front end TD:    
    const ecssg = ec2.SecurityGroup.fromSecurityGroupId(this, 'Imported ECS SG', importedSG)
    postgrestdbSG.sg.connections.allowFrom(
      ecssg,
      ec2.Port.tcp(5432),
    );
    //Importing ALB from output exported by other stack
    const importedalb = cdk.Fn.importValue("albarn");
    const importedalbsg = cdk.Fn.importValue("albsg");
    const imported_alb = elbv2.ApplicationLoadBalancer.fromApplicationLoadBalancerAttributes(this, 'imported ALB', {
      loadBalancerArn: importedalb,
      securityGroupId: importedalbsg
    })
    const ScoutTargetGroupdb = new elbv2.ApplicationTargetGroup(this, "scout-db-target-group",
      {
        port: 5432,
        vpc: defaultVpc,
        protocol: elbv2.ApplicationProtocol.HTTP,
        targetType: elbv2.TargetType.IP,
        targetGroupName: parameters.context.config.ScoutappContainername + '-' + parameters.context.config.ProjectEnvironment + '-' + 'postgrestdb'
      }
    );
    // Health check for containers to check they were deployed correctly 
    ScoutTargetGroupdb.configureHealthCheck({
      path: "/",
      protocol: elbv2.Protocol.HTTP,
    });
    const dblistener = imported_alb.addListener("alb-dbhttplistener", {
      open: true,
      port: 3000,
      protocol: elbv2.ApplicationProtocol.HTTP
    });
    dblistener.addTargetGroups("dbtargget group", {
      targetGroups: [ScoutTargetGroupdb],
    })

    // Fargate TaskDefinitions    
    const tdpostgrestdb = new ecs.FargateTaskDefinition(this, "tdpostgrestdb", {
      family: parameters.context.config.ProjectName + '-' + parameters.context.config.ProjectEnvironment + '-' + 'postgrest-ecs-task-def',
      memoryLimitMiB: 512,
      cpu: 256,
      taskRole: ecsTaskExecutionRole,
      executionRole: ecsTaskExecutionRole
    });
    const loggroup = new LogGroup(this, 'LogGroupscoutdb', {
      logGroupName: `/ecs/scout-app-postgrest-ecs-task-def`
    })
    const scoutimage = ssm.StringParameter.valueForStringParameter(
      this, 'scoutimage');
    const scoutapiimage = ssm.StringParameter.valueForStringParameter(
      this, 'scoutapi-image');
    const dbschema = ssm.StringParameter.valueForStringParameter(
      this, 'PGRST_DB_SCHEMA');
    const dburi = ssm.StringParameter.valueForStringParameter(
      this, 'PGRST_DB_URI');
    const jwtsecret = ssm.StringParameter.valueForStringParameter(
      this, 'PGRST_JWT_SECRET');
    const secretbase64 = ssm.StringParameter.valueForStringParameter(
      this, 'PGRST_SECRET_IS_BASE64');
    const authrole = ssm.StringParameter.valueForStringParameter(
      this, 'POSTGRES_AUTH_ROLE');
    const postgresdb = ssm.StringParameter.valueForStringParameter(
      this, 'POSTGRES_DB');
    const postgrespwd = ssm.StringParameter.valueForStringParameter(
      this, 'POSTGRES_PASSWORD');
    const dbanonrole = ssm.StringParameter.valueForStringParameter(
      this, 'PGRST_DB_ANON_ROLE');
    const postgresdbimage = ssm.StringParameter.valueForStringParameter(
      this, 'postgresdbimage');

    //Importing docker images from aws ecr
    const scoutdb_image = ecr.Repository.fromRepositoryArn(this, 'scoutdb repo', parameters.context.config.ScoutDbImage)
    //const scoutdb_image = ecr.Repository.fromRepositoryArn(this, 'scoutdb repo', 'arn:aws:ecr:us-east-1:321325872726:repository/ecr-metalapp')
    //ecs.EcrImage.fromEcrRepository(scoutdb_image, '2.0')

    //container definitions
    tdpostgrestdb.addContainer('postgrest-db', {
      image: ecs.ContainerImage.fromEcrRepository(scoutdb_image),
      memoryLimitMiB: 512,
      environment: { PGRST_DB_ANON_ROLE: dbanonrole, PGRST_DB_SCHEMA: dbschema, PGRST_DB_URI: dburi, PGRST_JWT_SECRET: jwtsecret, PGRST_SECRET_IS_BASE64: secretbase64, POSTGRES_AUTH_ROLE: authrole, POSTGRES_DB: postgresdb, POSTGRES_PASSWORD: postgrespwd },
      containerName: parameters.context.config.ProjectName + '-' + parameters.context.config.ProjectEnvironment + '-' + 'postgrest',
      portMappings: [{ containerPort: 3000, hostPort: 3000, protocol: ecs.Protocol.TCP }],
      essential: true,
      logging: ecs.LogDriver.awsLogs({
        streamPrefix: "ecs",
        logGroup: loggroup
      }),
    });
    // The ECS Service used for deploying tasks 
    const scoutpgecsservice = new ecs.FargateService(this, "scout-pg-ecs-service", {
      cluster: scoutdb_cluster,
      desiredCount: 0,
      taskDefinition: tdpostgrestdb,
      securityGroups: [ecssg],
      assignPublicIp: true,
      serviceName: parameters.context.config.ProjectName + '-' + parameters.context.config.ProjectEnvironment + '-' + 'pg-ecs-service',
    });
    scoutpgecsservice.attachToApplicationTargetGroup(ScoutTargetGroupdb)
    //Task Autoscaling based on memory and CPU usage
    const scalableTaget = scoutpgecsservice.autoScaleTaskCount({
      minCapacity: 0,
      maxCapacity: 2,
    });
    scalableTaget.scaleOnMemoryUtilization("ScaleUpMem-BackEnd", {
      targetUtilizationPercent: 75,
    });
    scalableTaget.scaleOnCpuUtilization("ScaleUpCPU-BackEnd", {
      targetUtilizationPercent: 75,
    });

  }
}
