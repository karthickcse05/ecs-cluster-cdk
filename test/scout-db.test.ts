import * as cdk from '@aws-cdk/core';
import * as AwsScoutDb from '../lib/scout-db';
import '@aws-cdk/assert/jest';
import { SynthUtils } from '@aws-cdk/assert';
import * as parameters from "../cdk.json"
import { Stack } from '@aws-cdk/core';
import * as elbv2 from '@aws-cdk/aws-elasticloadbalancingv2';

const myaws = { account: '321325872726', region: 'us-east-1' };


// Unit testing - fine-grained assertions on resources

/**
 * This is an example of a fine-grained test. This will check a single
 * aspect of the construct/stack.
 *
 * @see https://docs.aws.amazon.com/cdk/latest/guide/testing.html
 * @see https://aws.amazon.com/blogs/developer/testing-infrastructure-with-the-aws-cloud-development-kit-cdk/
 */


test('test to check the security group', () => {
  const stack = new cdk.App();
  const infrastructure = new AwsScoutDb.ScoutDBStack(stack, "ScoutDBTestStack",{
    env:myaws
  });
  expect(infrastructure).toHaveResource('AWS::EC2::SecurityGroup', {
    GroupName: parameters.context.config.ProjectName + '-' + parameters.context.config.ProjectEnvironment + '-' + 'db-sg'
  });
});

test('test to check the target group', () => {
    const stack = new cdk.App();
    const infrastructure = new AwsScoutDb.ScoutDBStack(stack, "ScoutDBTestStack",{
      env:myaws
    });
    expect(infrastructure).toHaveResource('AWS::ElasticLoadBalancingV2::TargetGroup', {
        Port:5432,
        Protocol:elbv2.ApplicationProtocol.HTTP,
        TargetType: elbv2.TargetType.IP,
        Name: parameters.context.config.ScoutappContainername + '-' + parameters.context.config.ProjectEnvironment + '-' + 'postgrestdb'
    });
  });

  test('test to check the task definition', () => {
  const stack = new cdk.App();
  const infrastructure = new AwsScoutDb.ScoutDBStack(stack, "ScoutDBTestStack",{
    env:myaws
  });
  expect(infrastructure).toHaveResource('AWS::ECS::TaskDefinition', {
    NetworkMode: "awsvpc",
    Memory: "512",
    Cpu: "256",
    Family: parameters.context.config.ProjectName + '-' + parameters.context.config.ProjectEnvironment + '-' + 'postgrest-ecs-task-def',
  });
});


test('test to check the Log group', () => {
  const stack = new cdk.App();
  const infrastructure = new AwsScoutDb.ScoutDBStack(stack, "ScoutDBTestStack",{
    env:myaws
  });
  expect(infrastructure).toHaveResource('AWS::Logs::LogGroup', {
    LogGroupName: "/ecs/scout-app-postgrest-ecs-task-def",
  });
});

test('test to check the ECS Service', () => {
  const stack = new cdk.App();
  const infrastructure = new AwsScoutDb.ScoutDBStack(stack, "ScoutDBTestStack",{
    env:myaws
  });
  expect(infrastructure).toHaveResource('AWS::ECS::Service', {
    ServiceName: parameters.context.config.ProjectName + '-' + parameters.context.config.ProjectEnvironment + '-' + 'pg-ecs-service',
    DesiredCount:0
  });
});

// SnapShot Testing
/**
 * This is an example of a snapshot test. This will essentially
 * check the "snapshot" to make sure it has everything we have defined,
 * and nothing more. It's essentially an integration test.
 *
 * @see https://docs.aws.amazon.com/cdk/latest/guide/testing.html
 * @see https://aws.amazon.com/blogs/developer/testing-infrastructure-with-the-aws-cloud-development-kit-cdk/
 */

 test("Test DB scout stack snapshot", () => {
  const stack = new Stack();
  const infrastructure = new AwsScoutDb.ScoutDBStack(stack, "ScoutDBTestStack",{
    env:myaws
  });

  expect(SynthUtils.toCloudFormation(infrastructure)).toMatchSnapshot();
});