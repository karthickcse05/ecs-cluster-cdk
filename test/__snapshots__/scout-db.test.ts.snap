// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test DB scout stack snapshot 1`] = `
Object {
  "Parameters": Object {
    "SsmParameterValuePGRSTDBANONROLEC96584B6F00A464EAD1953AFF4B05118Parameter": Object {
      "Default": "PGRST_DB_ANON_ROLE",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
    "SsmParameterValuePGRSTDBSCHEMAC96584B6F00A464EAD1953AFF4B05118Parameter": Object {
      "Default": "PGRST_DB_SCHEMA",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
    "SsmParameterValuePGRSTDBURIC96584B6F00A464EAD1953AFF4B05118Parameter": Object {
      "Default": "PGRST_DB_URI",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
    "SsmParameterValuePGRSTJWTSECRETC96584B6F00A464EAD1953AFF4B05118Parameter": Object {
      "Default": "PGRST_JWT_SECRET",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
    "SsmParameterValuePGRSTSECRETISBASE64C96584B6F00A464EAD1953AFF4B05118Parameter": Object {
      "Default": "PGRST_SECRET_IS_BASE64",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
    "SsmParameterValuePOSTGRESAUTHROLEC96584B6F00A464EAD1953AFF4B05118Parameter": Object {
      "Default": "POSTGRES_AUTH_ROLE",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
    "SsmParameterValuePOSTGRESDBC96584B6F00A464EAD1953AFF4B05118Parameter": Object {
      "Default": "POSTGRES_DB",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
    "SsmParameterValuePOSTGRESPASSWORDC96584B6F00A464EAD1953AFF4B05118Parameter": Object {
      "Default": "POSTGRES_PASSWORD",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
    "SsmParameterValuepostgresdbimageC96584B6F00A464EAD1953AFF4B05118Parameter": Object {
      "Default": "postgresdbimage",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
    "SsmParameterValuescoutapiimageC96584B6F00A464EAD1953AFF4B05118Parameter": Object {
      "Default": "scoutapi-image",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
    "SsmParameterValuescoutimageC96584B6F00A464EAD1953AFF4B05118Parameter": Object {
      "Default": "scoutimage",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": Object {
    "ImportedECSSGfromScoutDBTestStackimportedALBSecurityGroup677031D13000D250610D": Object {
      "Properties": Object {
        "Description": "Load balancer to target",
        "FromPort": 3000,
        "GroupId": Object {
          "Fn::ImportValue": "ecssg",
        },
        "IpProtocol": "tcp",
        "SourceSecurityGroupId": Object {
          "Fn::ImportValue": "albsg",
        },
        "ToPort": 3000,
      },
      "Type": "AWS::EC2::SecurityGroupIngress",
    },
    "LogGroupscoutdb0311533D": Object {
      "DeletionPolicy": "Retain",
      "Properties": Object {
        "LogGroupName": "/ecs/scout-app-postgrest-ecs-task-def",
        "RetentionInDays": 731,
      },
      "Type": "AWS::Logs::LogGroup",
      "UpdateReplacePolicy": "Retain",
    },
    "ScoutDBSGBackEnd3EB41EAB": Object {
      "Properties": Object {
        "GroupDescription": "Default/ScoutDBTestStack/ScoutDB-SG-BackEnd",
        "GroupName": "scout-prod-db-sg",
        "SecurityGroupEgress": Array [
          Object {
            "CidrIp": "0.0.0.0/0",
            "Description": "Allow all outbound traffic by default",
            "IpProtocol": "-1",
          },
        ],
        "SecurityGroupIngress": Array [
          Object {
            "CidrIp": "128.229.4.0/24",
            "Description": "Allow https traffic",
            "FromPort": 5432,
            "IpProtocol": "tcp",
            "ToPort": 5432,
          },
          Object {
            "CidrIp": "108.48.153.254/32",
            "Description": "Allow https traffic",
            "FromPort": 5432,
            "IpProtocol": "tcp",
            "ToPort": 5432,
          },
          Object {
            "CidrIp": "128.229.67.14/32",
            "Description": "Allow https traffic",
            "FromPort": 5432,
            "IpProtocol": "tcp",
            "ToPort": 5432,
          },
          Object {
            "CidrIp": "136.226.18.0/24",
            "Description": "Allow https traffic",
            "FromPort": 5432,
            "IpProtocol": "tcp",
            "ToPort": 5432,
          },
          Object {
            "CidrIp": "156.80.4.0/24",
            "Description": "Allow https traffic",
            "FromPort": 5432,
            "IpProtocol": "tcp",
            "ToPort": 5432,
          },
        ],
        "VpcId": "vpc-12345",
      },
      "Type": "AWS::EC2::SecurityGroup",
    },
    "ScoutDBSGBackEndfromScoutDBTestStackImportedECSSG61E262F95432B8C62E47": Object {
      "Properties": Object {
        "Description": "from ScoutDBTestStackImportedECSSG61E262F9:5432",
        "FromPort": 5432,
        "GroupId": Object {
          "Fn::GetAtt": Array [
            "ScoutDBSGBackEnd3EB41EAB",
            "GroupId",
          ],
        },
        "IpProtocol": "tcp",
        "SourceSecurityGroupId": Object {
          "Fn::ImportValue": "ecssg",
        },
        "ToPort": 5432,
      },
      "Type": "AWS::EC2::SecurityGroupIngress",
    },
    "importedALBSecurityGroupfrom000003000C5CA4B05": Object {
      "Properties": Object {
        "CidrIp": "0.0.0.0/0",
        "Description": "Allow from anyone on port 3000",
        "FromPort": 3000,
        "GroupId": Object {
          "Fn::ImportValue": "albsg",
        },
        "IpProtocol": "tcp",
        "ToPort": 3000,
      },
      "Type": "AWS::EC2::SecurityGroupIngress",
    },
    "importedALBalbdbhttplistener26DC6FBC": Object {
      "Properties": Object {
        "DefaultActions": Array [
          Object {
            "TargetGroupArn": Object {
              "Ref": "scoutdbtargetgroup8590F501",
            },
            "Type": "forward",
          },
        ],
        "LoadBalancerArn": Object {
          "Fn::ImportValue": "albarn",
        },
        "Port": 3000,
        "Protocol": "HTTP",
      },
      "Type": "AWS::ElasticLoadBalancingV2::Listener",
    },
    "scoutdbtargetgroup8590F501": Object {
      "Properties": Object {
        "HealthCheckPath": "/",
        "HealthCheckProtocol": "HTTP",
        "Name": "app-scout-prod-postgrestdb",
        "Port": 5432,
        "Protocol": "HTTP",
        "TargetType": "ip",
        "VpcId": "vpc-12345",
      },
      "Type": "AWS::ElasticLoadBalancingV2::TargetGroup",
    },
    "scoutpgecsserviceServiceE18424B9": Object {
      "DependsOn": Array [
        "importedALBalbdbhttplistener26DC6FBC",
      ],
      "Properties": Object {
        "Cluster": "app-scout-prod-ecs-cluster",
        "DeploymentConfiguration": Object {
          "MaximumPercent": 200,
          "MinimumHealthyPercent": 50,
        },
        "DesiredCount": 0,
        "EnableECSManagedTags": false,
        "HealthCheckGracePeriodSeconds": 60,
        "LaunchType": "FARGATE",
        "LoadBalancers": Array [
          Object {
            "ContainerName": "scout-prod-postgrest",
            "ContainerPort": 3000,
            "TargetGroupArn": Object {
              "Ref": "scoutdbtargetgroup8590F501",
            },
          },
        ],
        "NetworkConfiguration": Object {
          "AwsvpcConfiguration": Object {
            "AssignPublicIp": "ENABLED",
            "SecurityGroups": Array [
              Object {
                "Fn::ImportValue": "ecssg",
              },
            ],
            "Subnets": Array [
              "s-12345",
              "s-67890",
            ],
          },
        },
        "ServiceName": "scout-prod-pg-ecs-service",
        "TaskDefinition": Object {
          "Ref": "tdpostgrestdbBAB7FB89",
        },
      },
      "Type": "AWS::ECS::Service",
    },
    "scoutpgecsserviceTaskCountTarget30E4AC76": Object {
      "Properties": Object {
        "MaxCapacity": 2,
        "MinCapacity": 0,
        "ResourceId": Object {
          "Fn::Join": Array [
            "",
            Array [
              "service/app-scout-prod-ecs-cluster/",
              Object {
                "Fn::GetAtt": Array [
                  "scoutpgecsserviceServiceE18424B9",
                  "Name",
                ],
              },
            ],
          ],
        },
        "RoleARN": Object {
          "Fn::Join": Array [
            "",
            Array [
              "arn:",
              Object {
                "Ref": "AWS::Partition",
              },
              ":iam::321325872726:role/aws-service-role/ecs.application-autoscaling.amazonaws.com/AWSServiceRoleForApplicationAutoScaling_ECSService",
            ],
          ],
        },
        "ScalableDimension": "ecs:service:DesiredCount",
        "ServiceNamespace": "ecs",
      },
      "Type": "AWS::ApplicationAutoScaling::ScalableTarget",
    },
    "scoutpgecsserviceTaskCountTargetScaleUpCPUBackEnd6FE95615": Object {
      "Properties": Object {
        "PolicyName": "ScoutDBTestStackscoutpgecsserviceTaskCountTargetScaleUpCPUBackEndCA741356",
        "PolicyType": "TargetTrackingScaling",
        "ScalingTargetId": Object {
          "Ref": "scoutpgecsserviceTaskCountTarget30E4AC76",
        },
        "TargetTrackingScalingPolicyConfiguration": Object {
          "PredefinedMetricSpecification": Object {
            "PredefinedMetricType": "ECSServiceAverageCPUUtilization",
          },
          "TargetValue": 75,
        },
      },
      "Type": "AWS::ApplicationAutoScaling::ScalingPolicy",
    },
    "scoutpgecsserviceTaskCountTargetScaleUpMemBackEnd75C66927": Object {
      "Properties": Object {
        "PolicyName": "ScoutDBTestStackscoutpgecsserviceTaskCountTargetScaleUpMemBackEndA08EDB00",
        "PolicyType": "TargetTrackingScaling",
        "ScalingTargetId": Object {
          "Ref": "scoutpgecsserviceTaskCountTarget30E4AC76",
        },
        "TargetTrackingScalingPolicyConfiguration": Object {
          "PredefinedMetricSpecification": Object {
            "PredefinedMetricType": "ECSServiceAverageMemoryUtilization",
          },
          "TargetValue": 75,
        },
      },
      "Type": "AWS::ApplicationAutoScaling::ScalingPolicy",
    },
    "tdpostgrestdbBAB7FB89": Object {
      "Properties": Object {
        "ContainerDefinitions": Array [
          Object {
            "Environment": Array [
              Object {
                "Name": "PGRST_DB_ANON_ROLE",
                "Value": Object {
                  "Ref": "SsmParameterValuePGRSTDBANONROLEC96584B6F00A464EAD1953AFF4B05118Parameter",
                },
              },
              Object {
                "Name": "PGRST_DB_SCHEMA",
                "Value": Object {
                  "Ref": "SsmParameterValuePGRSTDBSCHEMAC96584B6F00A464EAD1953AFF4B05118Parameter",
                },
              },
              Object {
                "Name": "PGRST_DB_URI",
                "Value": Object {
                  "Ref": "SsmParameterValuePGRSTDBURIC96584B6F00A464EAD1953AFF4B05118Parameter",
                },
              },
              Object {
                "Name": "PGRST_JWT_SECRET",
                "Value": Object {
                  "Ref": "SsmParameterValuePGRSTJWTSECRETC96584B6F00A464EAD1953AFF4B05118Parameter",
                },
              },
              Object {
                "Name": "PGRST_SECRET_IS_BASE64",
                "Value": Object {
                  "Ref": "SsmParameterValuePGRSTSECRETISBASE64C96584B6F00A464EAD1953AFF4B05118Parameter",
                },
              },
              Object {
                "Name": "POSTGRES_AUTH_ROLE",
                "Value": Object {
                  "Ref": "SsmParameterValuePOSTGRESAUTHROLEC96584B6F00A464EAD1953AFF4B05118Parameter",
                },
              },
              Object {
                "Name": "POSTGRES_DB",
                "Value": Object {
                  "Ref": "SsmParameterValuePOSTGRESDBC96584B6F00A464EAD1953AFF4B05118Parameter",
                },
              },
              Object {
                "Name": "POSTGRES_PASSWORD",
                "Value": Object {
                  "Ref": "SsmParameterValuePOSTGRESPASSWORDC96584B6F00A464EAD1953AFF4B05118Parameter",
                },
              },
            ],
            "Essential": true,
            "Image": Object {
              "Fn::Join": Array [
                "",
                Array [
                  "321325872726.dkr.ecr.us-east-1.",
                  Object {
                    "Ref": "AWS::URLSuffix",
                  },
                  "/ecr-metalapp:latest",
                ],
              ],
            },
            "LogConfiguration": Object {
              "LogDriver": "awslogs",
              "Options": Object {
                "awslogs-group": Object {
                  "Ref": "LogGroupscoutdb0311533D",
                },
                "awslogs-region": "us-east-1",
                "awslogs-stream-prefix": "ecs",
              },
            },
            "Memory": 512,
            "Name": "scout-prod-postgrest",
            "PortMappings": Array [
              Object {
                "ContainerPort": 3000,
                "HostPort": 3000,
                "Protocol": "tcp",
              },
            ],
          },
        ],
        "Cpu": "256",
        "ExecutionRoleArn": "arn:aws:iam::321325872726:role/ecsTaskExecutionRole",
        "Family": "scout-prod-postgrest-ecs-task-def",
        "Memory": "512",
        "NetworkMode": "awsvpc",
        "RequiresCompatibilities": Array [
          "FARGATE",
        ],
        "TaskRoleArn": "arn:aws:iam::321325872726:role/ecsTaskExecutionRole",
      },
      "Type": "AWS::ECS::TaskDefinition",
    },
  },
}
`;
